tensorflow/core/framework/dataset.h

```diff
@@ -54,6 +56,102 @@ namespace tensorflow {
 // "tensorflow/core/graph/...".
 class GraphDefBuilder;
 class Node;
+class EparallaxTensorIndex;
+
+string ToString(std::vector<EparallaxTensorIndex*> indices);
+
+class EparallaxTensorIndex {
+ public:
+  EparallaxTensorIndex(string iterator_id,
+                       std::vector<EparallaxTensorIndex*>* parent_indices,
+                       int64 local_index) {
+    iterator_id_ = iterator_id;
```

Comment:
I think `iterator_id_` and `(parent_indices, local_index_)` can be interchangeable. What about removing `iterator_id_`?

Reply:
You're right, and `iterator_id_` is not necessary to uniquely identify each index.
But on second thought, `iterator_id_` is still very useful for identifying the type of the operator that produced the index. For example, it's convenient because we need to know the optype of the index to decide whether we can skip its parent index next time. (See `RecordFinished()` and `IsOneToManyOp()` method)